kubetools/kubernetes/config/cronjob.py

```diff
@@ -0,0 +1,77 @@
+import shlex
+
+from .container import make_container_config
+from .util import copy_and_update
+
+
+def make_cronjob_config(
+    config,
+    cronjob_name,
+    schedule,
+    containers,
+    labels=None,
+    annotations=None,
+    envvars=None,
+):
+    '''
+    Builds a Kubernetes cronjob configuration dict.
+    '''
+
+    labels = labels or {}
+    annotations = annotations or {}
+
+    # Build our container list
+    kubernetes_containers = []
+    for container_name, container in containers.items():
+        # Figure out the command
+        command = config['cronjobs'][cronjob_name]['containers'][container_name]['command']
+
+        if isinstance(command, str):
+            command = shlex.split(command)
+
+        # Get/create description
+        description = config.get('description', 'Run: {0}'.format(command))
+
+        # Attach description to annotations
+        annotations = copy_and_update(annotations, {
+            'description': description,
+        })
+
+        kubernetes_containers.append(make_container_config(
+            container_name, container,
+            envvars=envvars,
+            labels=labels,
+            annotations=annotations,
+        ))
+
+    # The actual cronjob spec
+    cronjob = {
+        'apiVersion': 'batch/v1',
+        'kind': 'CronJob',
+        'metadata': {
+            'name': cronjob_name,
+            'labels': labels,
+            'annotations': annotations,
+        },
+        'spec': {
+            'schedule': schedule,
+            'startingDeadlineSeconds': 10,
+            'jobTemplate': {
+                'spec': {
+                    'template': {
+                        'metadata': {
+                            'name': cronjob_name,
+                            'labels': labels,
+                            'annotations': annotations,
+                        },
+                        'spec': {
+                            'restartPolicy': 'OnFailure',
+                            'containers': kubernetes_containers,
+                        },
+                    },
+                },
+            },
```

Comment:
We should to add a line for `concurrencyPolicy` [configuration], it would be useful to be able to stop new cronjobs from cancelling out old ones that are long running (https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#concurrency-policy)

```suggestion
        'spec': {
            'schedule': schedule,
            'startingDeadlineSeconds': 10,
            'concurrencyPolicy': concurrency_policy,
            'jobTemplate': {
                'spec': {
                    'template': {
                        'metadata': {
                            'name': cronjob_name,
                            'labels': labels,
                            'annotations': annotations,
                        },
                        'spec': {
                            'restartPolicy': 'OnFailure',
                            'containers': kubernetes_containers,
                        },
                    },
                },
            },
```

Reply:
I added the `concurrencyPolicy` configuration